pkg/virt-handler/vm.go

```diff
@@ -67,6 +69,60 @@ type VMHandlerDispatch struct {
 	host          string
 }
 
+func (c *VMHandlerDispatch) getPodIPAddress() (string, error) {
+	// NB the libvirtd image is currently still in the host
+	// network namespace, so we need to figure out the
+	// default local outbound IP address it would use. Here
+	// we assume virt-handler is also in the same network
+	// namespace as libvirtd, and use this trick of binding
+	// to a local UDP socket lets us figure out the IP addr.
+	// If libvirtd were to move into a private network
+	// namespace, then we can instead switch to query
+	// the Status.PodIP field from the libvirtd POD.
```

Comment:
In short, yes - for now.

I've also summarized this in https://github.com/kubevirt/kubevirt/issues/196#issuecomment-306401959.
Right now - in this early development phase -it is okay to me to live with this fact that VMs are getting killed.
OTOH I'd work with k8s to find a way of how this could be solved on the k8s layer - as this problem might not be kubevirt specific.